src/trade/tradeDialog.es6

```diff
@@ -580,19 +581,29 @@ function init_state(available,root, dialog, symbol, contracts_for_spot) {
     /* contracts that are more not today must end at the market close time */
     const { value_date } = state.date_expiry;
     const is_today = !moment.utc(value_date).isAfter(moment.utc(), 'day');
+    const is_daily_contracts = state.duration_unit.array[0] && !hasIntradayUnit(state.duration_unit.array);
+
     if (!is_today) {
       state.date_expiry.today_times.disabled = true;
         trading_times_for(value_date, state.proposal.symbol)
           .then(function(times){
+            state.date_expiry.min_date = is_daily_contracts ? 1 : 0;
             const value_hour = times.close !== '--' ? times.close : '23:59:59';
             state.date_expiry.value_hour = moment.utc(value_hour, 'HH:mm:ss').format('HH:mm');
+            if (state.duration.value === 'Duration') {
```

Comment:
I don't think this will work with a translated 'Duration'?

Reply:
the `state.duration.value` taking the `HH:mm` format for endtime and when it goes duration it will changed to duration. i thought its the code for duration. let me check it again later

Reply:
i just check in different languages, the `state.duration.value` remain same which is `End time` and `Duration`